### ISSUE TITLE
wasm-opt --dfo crashes

### ISSUE THREAD

**Issue by hpohl** [WebAssembly/binaryen#2008]
Using it on [this file](https://s3.eu-central-1.amazonaws.com/trail-public/tmp/orig.wasm):
```
$ wasm-opt orig.wasm -o opt.wasm --dfo
Segmentation fault (core dumped)
```

**Comment by tlively** [comment]
Reduced reproducer: https://gist.github.com/tlively/4bc2a727cb956bc474b85d97edd9a9b6

**Comment by kripken** [comment]
Thanks! Looks like the same issue as #2007, that we don't give a proper error on IR not being flat when it is expected to be. Fix in #2009

**Comment by kripken** [comment]
Fixed by that PR.
